Formula/ksh93.rb

```diff
@@ -0,0 +1,23 @@
+class Ksh93 < Formula
+  desc "KornShell, ksh93"
+  homepage "https://github.com/ksh93/ksh#readme"
+  url "https://github.com/ksh93/ksh/archive/refs/tags/v1.0.0.tar.gz"
+  sha256 "54f1fcfea77ec55a639e08f0fe66a1fc918c762f176cad917ee93b44f511e1ef"
+  license "EPL-2.0"
+  head "https://github.com/ksh93/ksh.git", branch: "dev"
+
+  def install
+    system "bin/package", "verbose", "make"
+    system "bin/package", "verbose", "install", prefix
+    bin.install_symlink "ksh" => "ksh93"
+    bin.install_symlink "ksh" => "rksh"
+    man1.install_symlink "ksh.1" => "ksh93.1"
+    man1.install_symlink "ksh.1" => "rksh.1"
```

Comment:
I'm ok with your formula now, but I want to think a bit more about this snippet since it's really ugly. There has to be a nicer way to do this. But I suppose the simple way is good too...

Reply:
No problem!

And sorry to beat a dead horse, but let's get this done right (so we never have to go back to it ever again!)

About these symlinks ... a little thought on the matter reveals the current set of I made symlinks is insufficient.

There needs to be additional symlinks for `rksh93`, at least if we are to remain compatible with how things have worked in `ksh`-land for 30+ years.  Not doing so will inevitably result in astonishment and confusion for `ksh` users who use `ksh` on multiple systems (in all likelihood, this is 100% of `ksh` users these days.) 

I'll explain:

* When you ask for `ksh` you are going to get *some* variant of the Korn Shell, but the dialect is unspecified.  So, `ksh` gets you ksh88 on AIX, but it gets you ksh93 on Solaris, etc. 

  Thus, any **valid** `#!/bin/ksh` scripts absolutely **must** be coded to use *only* common features between these two dialects... and ksh93 is *not* exactly a superset of ksh88!  ksh88 has features not present in ksh93, and vice-versa, and these are non-trivial.

  For example, exported functions and aliases are commonly used in ksh88, but not present in ksh93. The ksh88 comparison operator is `=` and *not* `==` as in ksh93. Fall-through via `;&` is a ksh88 thing, and the list goes on and on for miles. Even for shells like `zsh` and `bash` that are somewhat compatible with at some subset of ksh features, there is no consensus on which ksh dialect they support for any individual feature. For example:

  ```ksh
  $ ksh93 -c 'echo $((1?X1=1:1))'
  1
  $ ksh88 -c 'echo $((1?X1=1:1))'  
  ksh: 1?X1=1:1: 0403-057 Syntax error
  $ bash -c 'echo $((1?X1=1:1))'
  1
  $ zsh -c 'echo $((1?X1=1:1))' 
  zsh:1: bad math expression: ':' expected
  ```
  So for this case `bash` acts like `ksh93`, but `zsh` acts like `ksh88`, yet for other functions, the opposite is true!

  Knowing which ksh dialect you need and having a way to specify that dialect is extremely important.

  This isn't limited to writing scripts - it bites interactive users, and in dozens of ways.  One example:

  * ksh88 uses `fc` for history, but `ksh93` uses `hist`, and these are not compatible, but in ksh93 `hist` is aliases to `fc` anyway, and it's commonly instead of `hist` since it's shorter to type, leading to frustrations if you find yourself in the "wrong" shell.
  <br>
  
  ```ksh
  $ A=1
  $ fc -N 2 -l
  ksh: fc: 0403-010 A specified flag is not valid for this command.
  ```

The problem with the missing `rksh93` symlinks is now apparent. A a portable ksh93 script that works everything now cannot be switched to use restricted mode anywhere at all in any portable way if `rksh` - a name which is insufficient to unambiguously specify the dialect required - can be either ksh93 or ksh88, without the corresponding `rksh93` link.

Without this extra symlink, a valid single character modification - putting an `r` in the shebang line - makes a working script into totally non-working script on a Mac, and also breaks the script on all the other primary systems I use, except Solaris, where it's fine.  Sorry I forgot this, and I'll commit the correction now.  This is not a hypothetical issue, it bit me today, hard.  The script in question was on an NFS export which I didn't have permission to modify, so I'd have been totally stuck if I hadn't been able to fix the ksh formula locally.

But, you you are right that it is ugly, and now it's uglier. \*sigh\*

A way to specify multiple targets for the symlink: `"ksh" => "ksh93", "rksh93", "rksh"` and `"ksh.1" => "ksh93.1", "rksh93.1", "rksh.1"` would be ideal, but it doesn't seem to be supported.

Reply:
Fixed in https://github.com/Homebrew/homebrew-core/pull/106033/commits/ff15aa0bf34c11269bbdff6284a1d613d7467e65